File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        d={'I':1,'V':5,'X':10,'L':50,'C':100,'D':500,'M':1000}
        res=0
        for i in reversed(s):
            val=d[i]
            if 4*val<res:
                res-=val
            else:
                res+=val
        return res

File: 0013-roman-to-integer/0013-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        Map<Character,Integer> a= new HashMap<Character,Integer>();
        a.put('I',1);
        a.put('V',5);
        a.put('X',10);
        a.put('L',50);
        a.put('C',100);
        a.put('D',500);
        a.put('M',1000);
        int ans=0;
        for(int i=0;i<s.length();i++){
            if(i!=s.length()-1 && (a.get(s.charAt(i))<a.get(s.charAt(i+1)))){
                ans=ans-a.get(s.charAt(i));
            }
            else{
                ans=ans+a.get(s.charAt(i));
            }
        }
        return ans;
    }
}